### PR TITLE
Multiple Trials for Reinforcement Learning Suggestion

### PR DESCRIPTION
**Pull request by DeeperMind**
Now the reinforcement learning suggestion supports spawning multiple trials in each iteration. Users can activate this function by specifying the `num_trials` suggestion parameter in StudyJob yaml file. 

Please notice that this multiple trials support is for extending exploration only, not for asynchronous update. The suggestion will use the average metrics of all the trials to calculate the policy gradient, which provides a more justified evaluation of the LSTM cell's internal state compared to using single trial. And the suggestion will not generate new candidates until all the previous ones are finished.

Fixes https://github.com/kubeflow/katib/issues/396

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/416)

### CODE REVIEW

File: pkg/suggestion/nasrl_service.py [hunk 1]
```diff
@@ -33,11 +35,13 @@ def __init__(self, request, logger):
         self.search_space = None
         self.opt_direction = None
         self.objective_name = None
+        self.num_trials = 1
```
Comment (hougangliu) [inline pkg/suggestion/nasrl_service.py]: api.GetSuggestionsRequest.request_number has defined num_trials, you need not add a new one in  suggestionParameters
Thanks!

Comment (andreyvelich) [reply pkg/suggestion/nasrl_service.py]: Thank you for notice! We will change it.

Comment (DeeperMind) [reply pkg/suggestion/nasrl_service.py]: Fixed. I removed num_trials from the suggestion parameters and used requestNumber instead.

**Comment by DeeperMind** [comment]
retest

**Pull request merged**